Simplenote/Classes/SPAboutViewController.swift

```diff
@@ -36,17 +36,17 @@ private extension SPAboutViewController {
 
     func setupDoneButton() {
         doneButton.translatesAutoresizingMaskIntoConstraints = false
-        doneButton.setTitle(NSLocalizedString("Done", comment: "Verb: Close current view"), for: UIControl.State.normal)
-        doneButton.setTitleColor(UIColor.white, for: UIControl.State.normal)
-        doneButton.setTitleColor(lightBlue, for: UIControl.State.highlighted)
-        doneButton.addTarget(self, action: #selector(onDoneTap(_:)), for: UIControl.Event.touchUpInside)
-        doneButton.contentHorizontalAlignment = UIControl.ContentHorizontalAlignment.right
+        doneButton.setTitle(NSLocalizedString("Done", comment: "Verb: Close current view"), for: .normal)
```

Comment:
just curious as one new to Swift: what's the resolution of `.normal`? Is there a kind of expanding context resolution? does it default to `this`/`self` and walk up the inheritance chain until it finds the nearest `.normal` property?

Reply:
Swift infers / knows the namespace, by checking against the method's signature API. Suppose you've got this codebase:

```
enum Test {
   case first
   case second
}

func doSomething(with test: Test) { /*...*/ }
```

You can do this:

```
doSomething(with: Test.first)
```

But since Swift already knows the parameter should be of `Test` type, it's okay for you to do...

```
doSomething(with: .first)
```

And that's one of the tiny reasons why we love Swift. Don't we?

Reply:
is that specifically because of `with:`?

I've seen similar systems in other languages that do things this way and even _with_ that sam keyword

Reply:
Oh nope, the "namespace resolution" capability is not tied up to the method's signature. You can literally use whatever in there, and it'll always work, regardless.

Relatively recent addition!